crmsh/pyshim.py

```diff
@@ -0,0 +1,21 @@
+import functools
+
+
+try:
+    from functools import cache
+except ImportError:
+    def cache(f):
+        cached_return_value = dict()
+
+        @functools.wraps(f)
+        def wrapper(*args, **kwargs):
+            nonlocal cached_return_value
+            key = (tuple(args), tuple(sorted(kwargs.items())))
+            try:
+                return cached_return_value[key]
+            except KeyError:
+                ret = f(*args, **kwargs)
+                cached_return_value[key] = ret
+                return ret
+
+        return wrapper
```

Comment:
crmsh already has `crmsh.cache` module, try to reuse it, or replace it

Reply:
functools.cache and crmsh.cache are quite different in features and designs.

1. functools.cache has simpler interface and does not support expiration and invalidation.
2. functools.cache is scoped by functions. crmsh.cache is global-scoped. A global-scoped cache requires selecting keys carefully, or limit its usage to avoid conflicts.

A feature-rich cache supporting all the use case of both functools.cache and crmsh.cache is usually implemented as a standalone library. It is an overkill to implement one in crmsh.

Reply:
Good.
Better to leave above points as comments in this py file?